src/frontend/Semantic_error.ml

```diff
@@ -152,6 +156,68 @@ module TypeError = struct
           (String.concat ~sep:"" lines)
           Fmt.(list UnsizedType.pp ~sep:comma)
           arg_tys
+    | IllTypedVariadicODE (name, arg_tys, args) ->
+        let types x = List.map ~f:(fun (_, t) -> t) x in
+        let optional_tol_args =
+          if Stan_math_signatures.is_variadic_ode_tol_fn name then
+            List.map
+              ~f:(fun (_, y) -> y)
+              Stan_math_signatures.variadic_ode_tol_arg_types
+          else []
+        in
+        let args_types = List.map ~f:(fun (_, t) -> t) args in
+        let generate_ode_sig =
+          [ UnsizedType.UFun
+              ( Stan_math_signatures.variadic_ode_mandatory_fun_args @ args
+              , ReturnType Stan_math_signatures.variadic_ode_fun_return_type )
+          ]
+          @ types Stan_math_signatures.variadic_ode_mandatory_arg_types
+          @ optional_tol_args @ args_types
+        in
+        let variadic_ode_generic_signature =
```

Comment:
IMO it's not clear why you're building these things, a comment describing your strategy throughout this section might make it more clear

Reply:
Added this comment: 

This function is used to generate the generic signature for variadic ODEs, i.e. with ... representing the variadic parts of the signature. This should be removed once a type representing variadic arguments is added. The generic signature is printed when there is a semantic error with one of the non-variadic arguments in the signature. If there is a mismatch in the variadic arguments, we print the non-generic expected signature (with explicit types for variadic args).